mosviz/viewers/mos_viewer.py

```diff
@@ -379,6 +379,74 @@ def add_data_for_testing(self, data):
         self._unpack_selection(data)
         return True
 
+    def add_data_for_jwst_testing(self, data):
+        """
+        Processes data message from the central communication hub.
+
+        Parameters
+        ----------
+        data : :class:`glue.core.data.Data`
+            Data object.
+        """
+
+        # Check whether the data is suitable for the MOSViz viewer - basically
+        # we expect a table of 1D columns with at least three string and four
+        # floating-point columns.
+        if data.ndim != 1:
+            QMessageBox.critical(self, "Error", "MOSViz viewer can only be used "
```

Comment:
Since this function should only ever be called in a non-interactive test environment, I think there shouldn't be any Qt messages here. Instead I think it's best just to raise an exception since that will cause the test to fail.